### PR TITLE
Bug/eui 1115 case progression aa

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-1115

### Change description ###

Copied and adapted navigation handler from CCD Consumer

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/store/effects/router.effect.ts [hunk 1]
```diff
@@ -22,8 +22,10 @@ export class RouterEffects {
   navigate$ = this.actions$.pipe(
     ofType(RouterActions.GO),
     map((action: RouterActions.Go) => action.payload),
-    tap(({ path, query: queryParams, extras }) => {
-      this.router.navigate(path, { queryParams, ...extras });
+    tap(({ path, query: queryParams, extras, callback, errorHandler }) => {
```
Comment (fazanki) [inline src/app/store/effects/router.effect.ts]: have learned lately that tap operator is usually the best used only for debugging purposes i.e to check memory leeks. you can use another map - also this observable is not completed as is not returning router.navigate

File: src/cases/containers/case-home/case-home.component.ts [hunk 2]
```diff
@@ -1,10 +1,113 @@
...
+    private titleService: ExUITitleService,
+    private alertService: AlertService,
+    private navigationNotifier: NavigationNotifierService,
+    private store: Store<fromFeature.State>,
+  ) { }
+
+  ngOnInit(): void {
+    this.navigationSubscription = this.navigationNotifier.navigation.subscribe(navigation => {
+      if (navigation.action) {
+        this.actionDispatcher(this.paramHandler(navigation));
```
Comment (fazanki) [inline src/cases/containers/case-home/case-home.component.ts]: this should move trigger effects

File: src/cases/containers/case-home/case-home.component.ts [hunk 3]
```diff
@@ -1,10 +1,113 @@
...
+
+  }
+
+  ngOnDestroy(): void {
+    if (this.navigationSubscription) {
+      this.navigationSubscription.unsubscribe();
+    }
+  }
+
+  paramHandler(navigation: any): GoActionParams {
```
Comment (fazanki) [inline src/cases/containers/case-home/case-home.component.ts]: this should move into effects and dispatch the .Go action directly within the statements

File: src/app/store/effects/router.effect.ts (hunk 1 above)
Comment (adnanakgun) [reply src/app/store/effects/router.effect.ts]: Can you redirect me to a source regarding tap-debugging please? What should we use instead?

The `return` makes sense, I'll fix that.

File: src/cases/containers/case-home/case-home.component.ts (hunk 2 above)
Comment (adnanakgun) [reply src/cases/containers/case-home/case-home.component.ts]: I'm not quite clear what you mean with this. The component only needs to trigger the `Go` action. The parameters and callbacks/error-handlers are prepared and passed on to the action. I don't understand where or why there is a need for an effect.

File: src/cases/containers/case-home/case-home.component.ts (hunk 3 above)
Comment (adnanakgun) [reply src/cases/containers/case-home/case-home.component.ts]: Same as above.

File: src/app/store/effects/router.effect.ts (hunk 1 above)
Comment (fazanki) [reply src/app/store/effects/router.effect.ts]: the source i have is in a book not with me.... i can explain tomorrow why is not a good idea... something to do with side effects... as if you resue the stream somewhere else i bring that side effect with it...

File: src/cases/containers/case-home/case-home.component.ts (hunk 2 above)
Comment (fazanki) [reply src/cases/containers/case-home/case-home.component.ts]: yeah fine but is a side effect and needs to be managed in the effect... so is just about to move this code to the effects

**Pull request merged**